### PR TITLE
status bar button (icon)

### PR DESCRIPTION
**Pull request by hnry**
The minimum to get a status bar button (icon) to appear...

I'm using this lib to learn rust and also cocoa, so please correct me if I'm misunderstanding something... Or if there is a contribution or styling guideline I am not following.

[Review on Reviewable](https://reviewable.io/reviews/servo/cocoa-rs/109)

### CODE REVIEW

**Comment by jdm** [comment]
This should probably update the package version as well so we can publish a new one.

File: src/appkit.rs [hunk 1]
```diff
@@ -2305,3 +2306,54 @@ impl NSScreen for id {
...
+
+pub const NSVariableStatusItemLength: CGFloat = -1.0;
+pub const NSSquareStatusItemLength: CGFloat = -2.0;
+
+pub trait NSStatusBar {
+    unsafe fn systemStatusBar(_: Self) -> id {
+        msg_send![class("NSStatusBar"), systemStatusBar]
+    }
+
+    unsafe fn statusItemWithLength_(self, length: CGFloat) -> id /* (NSStatusItem *) */;
```
Comment (jdm) [inline src/appkit.rs]: Why the _ here?

**Comment by hnry** [comment]
@jdm thanks for the quick response and note. I'm going to hold off on the version bump, I have a few more things I'd like to add. I just wanted to make sure what I had so far was acceptable.

**Comment by jdm** [comment]
It looks ok to me!

File: src/appkit.rs (hunk 1 above)
Comment (hnry) [inline src/appkit.rs]: @jdm the convention seems to be replace `:` with `_` (or methods that take an argument) so this seems correct? the method is `statusItemWithLength:`

the apple reference is [here] (https://developer.apple.com/library/prerelease/mac/documentation/Cocoa/Reference/ApplicationKit/Classes/NSStatusBar_Class/index.html#//apple_ref/occ/instm/NSStatusBar/statusItemWithLength:)

**Comment by hnry** [comment]
I am finished adding commits to this pull, let me know if I missed anything. 

NSStatusItem is complete, and NSStatusBar is almost complete (I am unsure how to access [thickness property](https://developer.apple.com/library/mac/documentation/Cocoa/Reference/ApplicationKit/Classes/NSStatusBar_Class/#//apple_ref/occ/instp/NSStatusBar/thickness)

Added a few more methods for NSImage, but that and NSButton are still missing a bunch, I hope it is ok to leave it as is with enough to just get a working status bar button/menu.

**Comment by jdm** [comment]
Thanks!

**Comment by jdm** [comment]
With regards to the `thickness` property, apparently those can be treated the same way as anything else: #110

**Comment by hnry** [comment]
Ok noted, thanks jdm for the help on this pull request, great projects always have great maintainers!

**Pull request merged**